frontend/src/utils/schedule-helpers.ts

```diff
@@ -141,3 +145,38 @@ export function scheduleHasClasses(schedule: Schedule): boolean {
   }
   return false;
 }
+
+/**
+ * Returns the course strings of every completed course in the given schedule.
+ * @param schedule the given schedule to parse
+ */
+export function getCompletedCourseStrings(schedule: DNDSchedule): string[] {
+  let fulfilled: string[] = [];
+
+  for (const y of schedule.years) {
+    const year: DNDScheduleYear = schedule.yearMap[y];
+
+    if (year.fall.classes.length !== 0) {
+      for (const course of year.fall.classes) {
+        fulfilled.push(course.subject + course.classId);
+      }
+    }
+    if (year.spring.classes.length !== 0) {
+      for (const course of year.spring.classes) {
+        fulfilled.push(course.subject + course.classId);
+      }
+    }
+    if (year.summer1.classes.length !== 0) {
+      for (const course of year.summer1.classes) {
+        fulfilled.push(course.subject + course.classId);
+      }
+    }
+    if (year.summer2.classes.length !== 0) {
+      for (const course of year.summer2.classes) {
+        fulfilled.push(course.subject + course.classId);
+      }
+    }
+  }
```

Comment:
Is it necessary to check that the length isn't 0? (since wouldn't the for each loop handle that)

Reply:
Thanks for catching this! That totally makes sense.